### PR TITLE
Fixes #2970 add eslint configurations to Docusaurus and play together with turborepo's linting

### LINKED ISSUE #2970: Configure ESLint for Docusaurus sub-project
Related to #2965 

After #2862, a number of ESLint errors showed up in the Docusaurus files. Eg

```
Parsing error: The keyword 'const' is reserved
```

### PR DESCRIPTION
**Pull request by cindyledev**
## Issue This PR Addresses
Fixes #2970 

## Type of Change

- [x] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

Before when we ran `pnpm lint`, we get only lint our tree and the Next.js frontend
```
PS C:\Users\lecin\Documents\repo\telescope> pnpm lint   

> @senecacdot/telescope@2.7.1 lint C:\Users\lecin\Documents\repo\telescope
> pnpm turbo run lint

• Packages in scope: @senecacdot/autodeployment, @senecacdot/eslint-config-telescope, @senecacdot/feed-discovery-service, @senecacdot/image-service, @senecacdot/planet-service, @senecacdot/posts-service, @senecacdot/search-service, @senecacdot/sso-service, @senecacdot/status-service, @senecacdot/telescope-frontend, dependency-discovery, migrate, parser, telescope-docs
• Running lint in 14 packages
@senecacdot/telescope-frontend:lint: cache hit, replaying output ea6f907d061ff4ab
@senecacdot/telescope-frontend:lint: 
@senecacdot/telescope-frontend:lint: > @senecacdot/telescope-frontend@3.0.0 lint C:\Users\lecin\Documents\repo\telescope\src\web
@senecacdot/telescope-frontend:lint: > pnpm eslint
@senecacdot/telescope-frontend:lint: 

 Tasks:    1 successful, 1 total
Cached:    1 cached, 1 total
  Time:    1.254s >>> FULL TURBO
```

With this PR, we will also lint the Docusaurus sub-project
```
PS C:\Users\lecin\Documents\repo\telescope> pnpm lint

> @senecacdot/telescope@2.7.1 lint C:\Users\lecin\Documents\repo\telescope
> pnpm turbo run lint

• Packages in scope: @senecacdot/autodeployment, @senecacdot/eslint-config-telescope, @senecacdot/feed-discovery-service, @senecacdot/image-service, @senecacdot/planet-service, @senecacdot/posts-service, @senecacdot/search-service, @senecacdot/sso-service, @senecacdot/status-service, @senecacdot/telescope-docs, @senecacdot/telescope-frontend, dependency-discovery, migrate, parser
• Running lint in 14 packages
@senecacdot/telescope-frontend:lint: cache hit, replaying output 7aa4bbab10a81473
@senecacdot/telescope-frontend:lint: 
@senecacdot/telescope-docs:lint: cache hit, replaying output 068f9f9922d75b3e
@senecacdot/telescope-docs:lint: 
@senecacdot/telescope-docs:lint: > @senecacdot/telescope-docs@0.0.1 lint C:\Users\lecin\Documents\repo\telescope\src\docs
@senecacdot/telescope-docs:lint: > pnpm eslint
@senecacdot/telescope-docs:lint: 
@senecacdot/telescope-frontend:lint: > @senecacdot/telescope-frontend@3.0.0 lint C:\Users\lecin\Documents\repo\telescope\src\web
@senecacdot/telescope-frontend:lint: > pnpm eslint
@senecacdot/telescope-frontend:lint: 

 Tasks:    2 successful, 2 total
Cached:    2 cached, 2 total
  Time:    1.284s >>> FULL TURBO
```

## Steps to test the PR
To test this PR locally:
1. Add my repository as a remote: `git remote add cindyledev https://github.com/cindyledev/telescope.git`
2. Fetch my branches: `git fetch cindyledev`
3. Checkout my branch: `git checkout issue-2970`
4. Install dependencies using pnpm in root (this will also install dependencies in src/docs: `pnpm install`
5. Lint: `pnpm lint`

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: tools/eslint/package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "eslint-plugin-jest": "25.3.4",
     "eslint-plugin-jest-playwright": "0.2.1",
     "eslint-plugin-jsx-a11y": "6.5.1",
+    "eslint-plugin-node": "11.1.0",
```
Comment (menghif) [inline tools/eslint/package.json:17]: Where do we use this?

Comment (cindyledev) [reply tools/eslint/package.json:17]: In PR #2853, we added a number of eslint overrides, and they addressed all the eslint errors we were getting for the Docusaurus app. Hence, the overrides in this PR are identical to what we had previously (just located in `src/docs` as opposed to `/`). 

For some reason, I am no longer getting any eslint errors for the Docusaurus app on master right now. I don't know if this PR is even needed...

Comment (humphd) [reply tools/eslint/package.json:17]: What if you introduce an eslint error, does it catch that?

Comment (cindyledev) [reply tools/eslint/package.json:17]: @menghif we use `eslint-plugin-node` here

https://github.com/Seneca-CDOT/telescope/blob/36f2307c949a5f943037cfcd979680430f1908f1/src/docs/.eslintrc.js#L25-L26

or else we get errors like these
2022-02-22 13_24_38-Window

**Review by humphd — APPROVED** [review]

**Review by tcvan0707 — DISMISSED** [review]

**Comment by aserputov** [comment]
@cindyledev we have three commits here.🤭

**Comment by cindyledev** [comment]
> @cindyledev we have three commits here.🤭 It's better to rebase I think.

three commits here but only one commit was merged into master ;)

**Pull request merged**